File: crates/libs/lib-core/src/models/spaceship.rs
use mongodb::bson::oid::ObjectId;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize, Default)]
pub struct Spaceship {
    pub _id: ObjectId,
    pub name: String,
    pub user_id: ObjectId,
    pub location_id: ObjectId,
    pub flown_out_at: Option<i64>, // timestamp
    pub flying: bool,
    pub system_id: ObjectId,
    pub planet_id: Option<ObjectId>,
}

File: api/models/spaceship.go
package models

import "github.com/google/uuid"

type Spaceship struct {
	ID         uuid.UUID `gorm:"type:uuid;default:gen_random_uuid()"`
	Name       string    `gorm:"not null"`
	UserID     uuid.UUID
	LocationID uuid.UUID
	Location   Location
	FlownOutAt int
	Flying     bool `gorm:"not null;default:false"`
	SystemID   uuid.UUID
	System     System
	PlanetID   uuid.UUID `gorm:"default:null"`
	Planet     Planet
}
